fedimint-api-client/src/api.rs

```diff
@@ -1168,42 +1177,57 @@ impl FederationPeerClientShared {
 #[derive(Debug)]
 struct FederationPeerClient<C> {
     client: JitTryAnyhow<C>,
-    shared: Arc<tokio::sync::Mutex<FederationPeerClientShared>>,
+    connection_state: Arc<tokio::sync::Mutex<FederationPeerClientConnectionState>>,
 }
 
 impl<C> FederationPeerClient<C>
 where
     C: JsonRpcClient + 'static,
 {
     pub fn new(peer_id: PeerId, url: SafeUrl, api_secret: Option<String>) -> Self {
-        let shared: Arc<_> = tokio::sync::Mutex::new(FederationPeerClientShared::new()).into();
+        let connection_state = Arc::new(tokio::sync::Mutex::new(
+            FederationPeerClientConnectionState::new(),
+        ));
 
         Self {
-            client: Self::new_jit_client(peer_id, url, api_secret, shared.clone()),
-            shared,
+            client: Self::new_jit_client(peer_id, url, api_secret, connection_state.clone()),
+            connection_state,
         }
     }
 
     fn new_jit_client(
         peer_id: PeerId,
         url: SafeUrl,
         api_secret: Option<String>,
-        shared: Arc<Mutex<FederationPeerClientShared>>,
+        connection_state: Arc<Mutex<FederationPeerClientConnectionState>>,
     ) -> JitTryAnyhow<C> {
         JitTryAnyhow::new_try(move || async move {
-            shared.lock().await.wait_and_inc_reconnect().await;
+            {
+                let mut cs = connection_state.lock().await;
+
+                if cs.last_connection_attempt_or.is_none() {
+                    info!(
+                    target: LOG_CLIENT_NET_API,
+                    peer_id = %peer_id,
+                    url = %url,
+                    "Connecting to peer...");
+                } else {
+                    debug!(
+                    target: LOG_CLIENT_NET_API,
+                    peer_id = %peer_id,
+                    url = %url,
+                    "Retrying connecting to peer...");
+                }
+
+                cs.wait().await;
```

Comment:
Would move this part into a function, as the current scope is getting long.

Reply:
Done.